nil-docs/zkllvm/getting-started/compiling-cpp-code.mdx

```diff
@@ -0,0 +1,31 @@
+# Compililing C++ code
+
+This tutorial shows how a `.cpp` file can be converted into a circuit IR.
+
+## Using `clang` from a pre-built binary
+
+If `clang` has been [**installed from a pre-built binary**](./installation#using-pre-built-binaries), use the following command.
+
+```bash
+clang-zkllvm -S -emit-llvm  path/to/circuit.cpp -target assigner -O1 -o circuit.ll
+```
+
+## Using `clang` from sources
+
+If `clang` has been [**installed from sources**](./installation#installing-from-sources), export the `CLANG_ZKLLVM` environmental variable.
+
+```bash
+export CLANG_ZKLLVM="path/to/clang"
+```
+
+:::info[`clang` location]
+
+`"path/to/clang"` is usually equal to `"zkllvm/build/libs/circifier/llvm/bin`.
+
+:::
+
+Afterward, call `$CLANG_ZKLLVM`.
+
+```bash
+$CLANG_ZKLLVM -S -emit-llvm path/to/circuit.cpp -target assigner -O1 -o circuit.ll
```

Comment:
During the experiments with `merkle_tree_poseidon.cpp`, I realized that we have to add standard library includes for the case if clang is built from sources. Otherwise any include like `<array>` will lead to compilation error. So please add:
```
-I/path/to/zkllvm//libs/stdlib/libcpp -I/path/to/zkllvm/libs/circifier/clang/lib/Headers -I/path/to/zkllvm/libs/stdlib/libc/include
```
Seems very complicated, but this time it is required, because circuits without any standard include are really exotic.

Reply:
thank you, added an admonition here